### PR TITLE
[Refactor RetinaNet] Step 9 unify NMSDecoder.

### PR DESCRIPTION
**Pull request by tanzhenyu**
1. This PR significantly reduces the logic for NMS decode.
2. It allows user to more easily fork and customize their own NMS decoder.
3. It unifies FasterRCNN and RetinaNet decoding logic.

### CODE REVIEW

File: keras_cv/layers/object_detection/nms_prediction_decoder.py [hunk 1]
```diff
@@ -15,115 +15,98 @@
...
         self,
         bounding_box_format,
-        anchor_generator,
-        classes=None,
-        suppression_layer=None,
-        box_variance=(0.1, 0.1, 0.2, 0.2),
+        from_logits,
+        iou_threshold=0.5,
+        confidence_threshold=0.5,
+        max_detections=10,
```
Comment (ianstenbit) [inline keras_cv/layers/object_detection/nms_prediction_decoder.py:46]: These defaults don't align with the docstrings, which say that defaults are 100 for each

File: keras_cv/layers/object_detection/nms_prediction_decoder.py [hunk 2]
```diff
@@ -15,115 +15,98 @@
...
+        self.from_logits = from_logits
+        self.iou_threshold = iou_threshold
+        self.confidence_threshold = confidence_threshold
+        self.max_detections = max_detections
+        self.max_detections_per_class = max_detections_per_class
         self.built = True
 
-    def call(self, images, predictions):
+    def call(self, box_pred, confidence_pred):
         """Accepts images and raw predictions, and returns bounding box predictions.
```
Comment (ianstenbit) [inline keras_cv/layers/object_detection/nms_prediction_decoder.py:60]: nit: this docstring is now out-of-date

**Review by ianstenbit — APPROVED** [review]
LGTM mod a few nits, and as long as the two example scripts are still WAI.

Thank you!

File: keras_cv/layers/object_detection/nms_prediction_decoder.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/layers/object_detection/nms_prediction_decoder.py]: Good catch. Done

File: keras_cv/layers/object_detection/nms_prediction_decoder.py (hunk 2 above)
Comment (tanzhenyu) [reply keras_cv/layers/object_detection/nms_prediction_decoder.py:60]: Good catch. Fixed.

File: keras_cv/layers/__init__.py [hunk 3]
```diff
@@ -23,8 +23,7 @@
 from keras_cv.layers.mbconv import MBConvBlock
 from keras_cv.layers.object_detection3d.voxelization import DynamicVoxelization
 from keras_cv.layers.object_detection.anchor_generator import AnchorGenerator
-from keras_cv.layers.object_detection.nms_prediction_decoder import NmsPredictionDecoder
-from keras_cv.layers.object_detection.non_max_suppression import NonMaxSuppression
+from keras_cv.layers.object_detection.nms_prediction_decoder import NmsDecoder
```
Comment (innat) [inline keras_cv/layers/__init__.py:26]: NMSDecoder instead of NmsDecoder?

**Pull request merged**